mariadb/config.json

```diff
@@ -25,7 +25,7 @@
   "schema": {
     "databases": ["str"],
     "logins": [{ "username": "str", "password": "password" }],
-    "rights": [{ "username": "str", "database": "str" }]
+    "rights": [{ "username": "str", "database": "str", "privileges": "str?" }]
```

Comment:
Use ["list(X|Y|Z)"], we should not allow making mistakes

Reply:
So I can't do this currently. The config is expecting the user to provide a string with one or more privileges separated by commas like this `SELECT, CREATE, UPDATE`. I could only do a `list(...)` config if I changed it to this:

```json
"rights": [{ "username": "str", "database": "str", "privileges": ["list(X|Y|Z)?"] }]
```

This is the configuration I wanted to do but it would be a breaking change. All existing configs would need to be adjusted like this to pass validation now and continue to work as expected
```diff
rights:
  - username: admin
    database: testdb
+  privileges: []
```

I suppose I can do a long `match(...)` configuration though so I'll get to work on that for now.

Reply:
If there is a way to add an optional array config that can be omitted let me know. I would prefer users configured this like this, think it would be nicer:

```yaml
rights:
  - username: admin
    database: testdb
    privileges:
      - SELECT
      - CREATE
      - UPDATE
```

I just didn't want to introduce a breaking change

Reply:
Added a match statement. Tested it by pulling it into `/addons` as a local addon and trying out various good and bad configs. Seems to correctly enforce the syntax we're looking for now. Let me know if that works.

Reply:
`"privileges": ["list(x|y|z)?"]` should work. Otherwise, it's a bug which we should fix on the supervisor and release the add-on after the bugfix.

Reply:
Ok makes sense. So yea looks like that configuration isn't working properly. I made a simple test addon with this for `schema` and `options`:

```json
{
	"options": {
		"log_level": "info"
	},
  "schema": {
		"log_level": "str",
		"privileges": ["list(x|y|z)?"]
	}
}
```

If I just `log_level` whatever I type won't be saved and will just revert because its invalid. But as soon as I add `privileges: []` to my config then its all good and it saves fine. Basically it treats the elements inside the array as option but the field itself becomes mandatory within that object. I can set it to `[]` but I can't leave it out.

Reply:
Added a [PR](https://github.com/home-assistant/supervisor/pull/2752) to supervisor to add that capability. Will now update this to use it

Reply:
Ok just updated it to use a list properly now. Also tweaked the DOCS in light of the change and modified the example to show a read-only user for completeness.